src/Webapi/Dom/Webapi__Dom__HtmlButtonElement.res

```diff
@@ -0,0 +1,55 @@
+/**
+ * Spec: https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element
+ */
+
+type t = Dom.htmlButtonElement
+
+include Webapi__Dom__EventTarget.Impl({
+  type t = t
+})
+include Webapi__Dom__Node.Impl({
+  type t = t
+})
+include Webapi__Dom__Element.Impl({
+  type t = t
+})
+include Webapi__Dom__HtmlElement.Impl({
+  type t = t
+})
+
+// common form element attributes
+@return(nullable) @get
+external form: t => option<Dom.htmlFormElement> = "form"
+@get external labels: t => Dom.nodeList = "labels"
+
+@get external name: t => string = "name"
+@set external setName: (t, string) => unit = "name"
+
+@get external disabled: t => bool = "disabled"
+@set external setDisabled: (t, bool) => unit = "disabled"
+
+@get external value: t => string = "value"
+@set external setValue: (t, string) => unit = "value"
+
+/* Validation */
+@get external willValidate: t => bool = "willValidate"
+@get external validity: t => Webapi__Dom__ValidityState.t = "validity"
+@get external validationMessage: t => string = "validationMessage"
+@send external setCustomValidity: (t, string) => unit = "setCustomValidity"
+@send external checkValidity: t => bool = "checkValidity"
+@send external reportValidity: t => bool = "reportValidity"
```

Comment:
only question I have left is that this is repeated in a few places, is it worth pulling out?

Reply:
I was wondering that myself, actually. Every form element has some properties they share.

I doubt the form APIs will need much maintenance, though, and there aren't really very many places it's duplicated compared to other places the `Impl` technique is used, so I think leave it as is for now.